pkg/cephfs/nodeserver.go

```diff
@@ -308,6 +309,89 @@ func (ns *NodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstag
 	return &csi.NodeUnstageVolumeResponse{}, nil
 }
 
+func (ns *NodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
+
+	var err error
+	targetPath := req.GetVolumePath()
+	if targetPath == "" {
+		err = fmt.Errorf("targetpath %v is empty", targetPath)
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	volID := req.GetVolumeId()
+	/* TODO: Map the volumeID to the targetpath.
+
+	we need secret to connect to the ceph cluster to get the volumeID from volume
+	Name, however `secret` field/option is not available  in NodeGetVolumeStats spec,
+	Below issue covers this request and once its available, we can do the validation
+	as per the spec.
+	https://github.com/container-storage-interface/spec/issues/371
+
+	*/
+
+	isMnt, err := isMountPoint(targetPath)
+
+	if err != nil {
+		klog.Errorf("stat failed on path: %v, error: %v", targetPath, err)
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	if isMnt {
+		klog.Infof("cephfs: volume %s is already mounted to %s", volID, targetPath)
+		cephfsProvider := volume.NewMetricsStatFS(targetPath)
+		volMetrics, volMetErr := cephfsProvider.GetMetrics()
+		if volMetErr != nil {
+			return nil, status.Error(codes.Internal, volMetErr.Error())
+		}
+
+		available, ok := (*(volMetrics.Available)).AsInt64()
+		if !ok {
+			klog.Errorf("failed to fetch available bytes")
+		}
+		capacity, ok := (*(volMetrics.Capacity)).AsInt64()
```

Comment:
As per spec `total` is a REQUIRED field, so if there are errors in fetching Capacity or Inodes totals, we should error out?

Reply:
we can return an error if inodes and total fails.

Reply:
Done.